### ISSUE TITLE
RB25s does not switch into automatic mode in case of failsafe , channel 14 = -100%

### ISSUE THREAD

**Issue by Kreisflieger** [FrSkyRC/RF_Feedback_Group#26]
I am using a RB25S in Quick-Mode with an Archer Plus R8 connected via FBus. I use a X18SE with Ethos 1.5.3.
Firmware of RB25S, Archer Plus R8 and RB25-LUAS are all on latest versions.
To setup stabilization I set channel 13 (gain) on a pot and channel 14 (stab-mode) on a switch.
Everything works fine in Stabilization and Automatic Mode, as expected. 

Now I want to activate automatic mode when the receiver goes into failsafe.
Therefore I set channel 14 in the LUA config page 3/3 (failsafe settings) to -100%. Gain is set to 70% ( appr. 1750 micro sec. ) This should activate automatic mode with a good gain setting.
Unfortunately, the RB25S does not switch to automatic mode when receiver is in failsafe.

Checking the servo output of the RB25S on channel 14, it is switched to -100% which equals automatic mode.
But the RB25S is still in stabilization “off mode”.
I also set the receiver channel 14 to -100%. No success, either.
So I assume that there is a missing logical connection somewhere in the setup: Ethos, config LUA or RB25 firmware.
As known, all current S-receivers switch into automatic mode, if failsafe on channel 14 is set to 
-100%. 
Why not the RB25S ??

Kind regards

**Comment by Kreisflieger** [comment]
Hope, the RF team will now take care about this issue.

**Comment by SmartTommy** [comment]
[RBX5.zip](https://github.com/user-attachments/files/18305380/RBX5.zip)

Please conduct a comprehensive test to see if it can meet your needs. 

use the ethos 1.6.0 and TX & RX 3.0.1 and the new LUA(included in the packet)

**Comment by Kreisflieger** [comment]
@ SmartTommy
Thanks for providing the files.
I tested RBX5.zip with:
X18RS, Ethos 1.6.0, TD-ISRM RS 3.0.1
Archer Plus R8, 3.01, connected via FBus
RB25S with your provided Firmware: RB25S_3.0.1RC2.frsk and LUA RB25_S ( RBx5)
See Info:
 
 
![1](https://github.com/user-attachments/assets/6d3ab3f5-ac71-466d-8f67-efaeb9e18f9f)
![2](https://github.com/user-attachments/assets/55362c25-f247-4896-8c6e-623b33671efb)

Using the device config:
 
![3](https://github.com/user-attachments/assets/2b2ee694-f8fc-43bb-baf3-db066cbbcc31)

![4](https://github.com/user-attachments/assets/441cddee-c44f-46a7-ad2c-d55f94477490)

Selecting RB25S calibration does not show the drawing of the RB25S as provided by the
 cali folder:
 
![a](https://github.com/user-attachments/assets/26b7a68d-65ef-4898-a8ae-8aa43d0f8562)

But the calibration does work
 

Test result of the RB25S is  as expected
As the failsafe for channel 14 was still set ( on  page 1-3 of RB25(S) ) to 890 micro sec, switching off the transmitter, the RB25S switched instantly into automatic mode, perfect ! That what I was looking for.

But, I was not able to alter the failsafe setting of other channels. As soon as I click on the field: **Record current value**, I get a LUA warning **red triangle**  and the value of the channel resets to the before value.
 
![5](https://github.com/user-attachments/assets/64110d65-bfe6-4ed5-b491-3a60e6c2205c)

Clicking on the triangle shows:
 
![6](https://github.com/user-attachments/assets/7d6b4433-f044-4651-a8eb-680584a5472e)

To sum up my findings:
-	The general function of the RB25S for flying is now as expected.
LUAS have to be improved:
-	In calibration mode, the drawing of the RB25S should be shown
-	Setting the failsafe values should not create a red triangle warning and the new failsafe value should be stored

Hope my findings are helpful for you to improve the LUAs.

Many regards 
Stephan

**Comment by Kreisflieger** [comment]
While I was still testing the new version I got an extended LUA Script warning:

![7](https://github.com/user-attachments/assets/0e5955dc-733e-4770-b757-b4db9a84fbab)

**Comment by Nicholas-Luoyi** [comment]
Fixed the first error. But I cannot reproduce the second error

**Comment by Kreisflieger** [comment]
Thanks Nicolas,

if you can provide the updated LUAs I am happy to further test.

Regards

**Comment by Nicholas-Luoyi** [comment]
Will upload 2.1.2 later. Please update it in Suite

**Comment by Nicholas-Luoyi** [comment]
> go to page 3-3 of config, directly type in a value in the left hand microsecond field, e.g. 1000

I guess you mean right hand?

**Comment by Kreisflieger** [comment]
yes, sorry for the mix up.

![8](https://github.com/user-attachments/assets/633d1387-2c2e-4ffe-99aa-6671becb859b)

and the corresponding red triangle message:
![9](https://github.com/user-attachments/assets/42db9484-30b9-49c8-b6e3-f0385065bf26)

Kind regards Stephan

**Comment by Nicholas-Luoyi** [comment]
Still cannot reproduce. But I can propose a fix. Please try.

[RB25_S.zip](https://github.com/user-attachments/files/18329380/RB25_S.zip)

**Comment by Kreisflieger** [comment]
tested your latest version:

if I change the channel 1 value field ( e.g. to 1000 ) with the stick in middle position and pressing the "record current value" the value field jumps to 1500, no red triangel.

This is not too helpful but it avoids the red triangle.

But instead may I suggest to change the value field into a **read only field** and display the current value of that channel to be able to set the channel to the desired value, when pressing "Record current value"

Kind regards  Stephan

**Comment by Kreisflieger** [comment]
Dear Nicholas,
to further dig into this behavour, can you please send the RB25 firmware, which you are using for your test/ LUA development.

Then we will be sure that I have the the correct version.

Thanks in advance   Stephan

**Comment by Nicholas-Luoyi** [comment]
They are independent from each other. I am responsible for Lua scripts. And @SmartTommy is responsible for the device firmware

**Comment by Kreisflieger** [comment]
Thanks, good to know.

I will continue to test, as I do not understand why my X18RS is behaving differently to your setup.

**Comment by SmartTommy** [comment]
> Thanks, good to know.
> 
> I will continue to test, as I do not understand why my X18RS is behaving differently to your setup.

[RB35_3.0.1_RC1.zip](https://github.com/user-attachments/files/18344189/RB35_3.0.1_RC1.zip)

**Comment by Kreisflieger** [comment]
Hello Nicolas,

success, the RB25 is working ok now !!!

I found the reason, why I had problems with the value field on page 3-3 of RB25(S) config:

if I change the value e.g. of channel 1 from 1500 to 1000, the field is hightlighted in yellow. If I now press "Record current value" I get the LUA error.

But if I confirm my setting to 1000 with the enter button, the the value is stored without any errors.

**So it was more or less a user error ! Sorry for that.**

But maybe you can implement a functionality, so no other users will experience my fault as well.

Thank you very much for your continous support to improve the FrSky system.

Many regards Stephan
